src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Properties.jl

```diff
@@ -138,6 +138,53 @@ function _projective_jacobian_criterion(P::AbsProjectiveScheme{<:Ring, <:MPolyQu
   return is_one(saturation(sing_locus, irrelevant_ideal))
 end
 
+@doc raw"""
+    singular_locus(P::AbsProjectiveScheme) -> AbsProjectiveScheme
+
+Given an equidimensional projective scheme, returns the possibly
+nonreduced subscheme of projective space describing the singular locus.
+
+The algorithm first checks for equidimensionality, which can be expensive.
+If you already know that the scheme is equidimensional, then you can
+avoid recomputing that by writing
+  `set_attribute!(P, :is_equidimensional, true)`
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = graded_polynomial_ring(QQ, ["x", "y", "z"])
+(Graded multivariate polynomial ring in 3 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y, z])
+
+julia> I = ideal(R, x^2 + y^2)
+Ideal generated by
+  x^2 + y^2
+
+julia> X = proj(I)
+Projective scheme
+  over rational field
+defined by ideal (x^2 + y^2)
+
+julia> singular_locus(X)
+Projective scheme
+  over rational field
+defined by ideal (y, x)
+```
+"""
+function singular_locus(P::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing})
+  if !(is_equidimensional(P))
+    throw(NotImplementedError(
+      :is_smooth,
+      "projective Jacobian criterion only implemented when scheme is equidimensional. Instead, you could try `singular_locus(covered_scheme(X))`"
+    ))
+  end
```

Comment:
```suggestion
function singular_locus(P::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing}; check::Bool=true)
  @check begin
    if !(is_equidimensional(P))
      throw(NotImplementedError(
        :is_smooth,
        "projective Jacobian criterion only implemented when scheme is equidimensional. Instead, you could try `singular_locus(covered_scheme(X))`"
      ))
    end
  end
```
The equidimensionality-check is tentatively expensive, so we probably want a way to turn it off. This should do the job.

Reply:
Indeed..